sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -206,7 +200,7 @@ class Analyzer(
    */
   val postHocResolutionRules: Seq[Rule[LogicalPlan]] = Nil
 
-  lazy val batches: Seq[Batch] = Seq(
+  def batches: Seq[Batch] = Seq(
```

Comment:
nit: could you add `override` here? https://github.com/databricks/scala-style-guide#override-modifier